### PR TITLE
Ahmad/WEBREL-1630/Convert vite to webpack

### PR DESCRIPTION
**Pull request by ahmadtaimoor-deriv**

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/* eslint-disable global-require */
+
```
Comment (niloofar-deriv) [inline .storybook/main.js]: What was the reason behind changing the format of this file? 🤔

File: package.json [hunk 2]
```diff
@@ -4,30 +4,38 @@
...
         "@tanstack/react-query": "^4.35.7",
         "@tanstack/react-query-devtools": "^4.35.7",
         "class-variance-authority": "^0.7.0",
         "clsx": "^2.0.0",
         "i18next": "^23.5.1",
+        "jest-chain": "^1.1.6",
+        "jest-environment-jsdom": "^29.7.0",
+        "jest-extended": "^4.0.2",
+        "mini-css-extract-plugin": "^2.7.6",
+        "nx": "^17.0.3",
```
Comment (niloofar-deriv) [inline package.json]: Do we need nx? 🤔  This is not a monorepo

File: package.json [hunk 3]
```diff
@@ -4,30 +4,38 @@
...
     "dependencies": {
+        "@babel/preset-react": "^7.23.3",
         "@deriv/api-types": "^1.0.126",
-        "@deriv/deriv-api": "^1.0.13",
         "@radix-ui/react-dialog": "^1.0.5",
         "@radix-ui/react-dropdown-menu": "^2.0.6",
         "@radix-ui/react-icons": "^1.3.0",
         "@radix-ui/react-slot": "^1.0.2",
         "@radix-ui/react-tooltip": "^1.0.7",
+        "@squoosh/lib": "^0.5.3",
```
Comment (henry-deriv) [inline package.json]: what's this for

File: src/components/common/loading/__test__/loading.test.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // Test for passing the github actions
 // TODO should update this test later
-import { describe, expect, test } from 'vitest';
 import { render, screen } from '@testing-library/react';
+
```
Comment (henry-deriv) [inline src/components/common/loading/__test__/loading.test.tsx]:
```suggestion
```

File: src/main.tsx [hunk 5]
```diff
@@ -1,11 +1,13 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 import App from './App';
+
```
Comment (henry-deriv) [inline src/main.tsx]:
```suggestion
```

**Comment by balakrishna-deriv** [comment]
Seems like the tests are failing. Can you please check?

File: .storybook/main.js (hunk 1 above)
Comment (balakrishna-deriv) [reply .storybook/main.js]: yeah, why js?

File: package.json [hunk 6]
```diff
@@ -4,30 +4,38 @@
...
         "@tanstack/react-query-devtools": "^4.35.7",
         "class-variance-authority": "^0.7.0",
         "clsx": "^2.0.0",
         "i18next": "^23.5.1",
+        "jest-chain": "^1.1.6",
+        "jest-environment-jsdom": "^29.7.0",
+        "jest-extended": "^4.0.2",
+        "mini-css-extract-plugin": "^2.7.6",
+        "nx": "^17.0.3",
+        "path": "^0.12.7",
```
Comment (balakrishna-deriv) [inline package.json]: path should be in devdependecy

File: package.json [hunk 7]
```diff
@@ -4,30 +4,38 @@
...
+        "@storybook/addon-postcss": "^2.0.0",
         "@tanstack/react-query": "^4.35.7",
         "@tanstack/react-query-devtools": "^4.35.7",
         "class-variance-authority": "^0.7.0",
         "clsx": "^2.0.0",
         "i18next": "^23.5.1",
+        "jest-chain": "^1.1.6",
+        "jest-environment-jsdom": "^29.7.0",
+        "jest-extended": "^4.0.2",
+        "mini-css-extract-plugin": "^2.7.6",
```
Comment (balakrishna-deriv) [inline package.json]: all these should be in dev dependencies

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
+import path from 'path';
```
Comment (balakrishna-deriv) [inline webpack.config.js]: Can you update the file to .ts?

File: package.json [hunk 9]
```diff
@@ -4,30 +4,38 @@
...
+        "test:watch": "jest --watch",
+        "test:coverage": "jest --coverage",
         "storybook": "storybook dev -p 6006",
         "build-storybook": "storybook build",
         "prepare": "husky install"
     },
     "dependencies": {
+        "@babel/preset-react": "^7.23.3",
         "@deriv/api-types": "^1.0.126",
-        "@deriv/deriv-api": "^1.0.13",
```
Comment (balakrishna-deriv) [inline package.json:20]: We don't use `@deriv/deriv-api` anymore?

Comment (niloofar-deriv) [reply package.json:20]: He just moved that to devdependencies 🤔

File: src/components/common/loading/__test__/loading.test.tsx [hunk 10]
```diff
@@ -1,7 +1,7 @@
 // Test for passing the github actions
```
Comment (niloofar-deriv) [inline src/components/common/loading/__test__/loading.test.tsx]: You can remove this test file, I just added that before for passing the build 🙏

File: src/main.tsx [hunk 11]
```diff
@@ -1,11 +1,13 @@
...
 import APIProvider from 'Utils/websocket/APIProvider';
-import 'Styles/main.css';
 import 'Translations/i18n.ts';
+import 'Styles/main.css';
 
-ReactDOM.createRoot(document.getElementById('root')!).render(
+const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
```
Comment (niloofar-deriv) [inline src/main.tsx]: Do u need to change this? 🤔

File: package.json (hunk 2 above)
Comment (ahmadtaimoor-deriv) [reply package.json]: removed thanks

File: package.json (hunk 3 above)
Comment (ahmadtaimoor-deriv) [reply package.json]: It is image compression library . Removed now as we are not using :).

File: package.json (hunk 9 above)
Comment (ahmadtaimoor-deriv) [reply package.json:20]: It is added , thanks

File: .storybook/main.js (hunk 1 above)
Comment (ahmadtaimoor-deriv) [reply .storybook/main.js]: Changed to ts  now . It was transformed for some testing purpose

File: webpack.config.js (hunk 8 above)
Comment (ahmadtaimoor-deriv) [reply webpack.config.js]: updated the file to ts now

File: src/main.tsx (hunk 11 above)
Comment (ahmadtaimoor-deriv) [reply src/main.tsx]: reverted to old one . It was changed for testing purpose :)

File: .storybook/main.ts [hunk 12]
```diff
@@ -1,19 +1,35 @@
...
-        '@storybook/addon-interactions',
-    ],
-    framework: {
-        name: '@storybook/react-vite',
-        options: {},
-    },
-    docs: {
-        autodocs: 'tag',
+    framework: '@storybook/react-webpack5',
+    addons: ['@storybook/addon-links', '@storybook/addon-essentials'],
```
Comment (henry-deriv) [inline .storybook/main.ts]: what about @storybook/addon-interactions, and @storybook/addon-onboarding?

I dont know what they are exactly but should they be put back as before?

File: .storybook/main.ts [hunk 13]
```diff
@@ -1,19 +1,35 @@
...
+                    },
+                },
+            ],
+            include: path.resolve(dirname, '../'),
+        });
+        if (config.resolve) {
+            config.resolve.plugins = [new TsconfigPathsPlugin()];
+        }
+
+        return config;
```
Comment (henry-deriv) [inline .storybook/main.ts:41]: also i believe you are missing
```
docs: {
  autodocs: 'tag',
}
```

Comment (ahmadtaimoor-deriv) [reply .storybook/main.ts:41]: Added now, Thanks

File: .storybook/main.ts (hunk 12 above)
Comment (ahmadtaimoor-deriv) [reply .storybook/main.ts]: @storybook/addon-interactions : It allows debugging. 
@storybook/addon-onboarding: This addon provides a guided tour in some of Storybook's features . Not sure if we need it .

Added , Thanks

File: setupTests.js [hunk 14]
```diff
@@ -0,0 +1 @@
+require('@testing-library/jest-dom');
```
Comment (henry-deriv) [inline setupTests.js]: should remove this file right?

**Review by henry-deriv — APPROVED** [review]
lgtm

**Review by niloofar-deriv — APPROVED** [review]

**Review by balakrishna-deriv — DISMISSED** [review]

**Review by balakrishna-deriv — APPROVED** [review]

**Pull request merged**